File: Multiply Strings.cpp
class Solution {
public:
    string multiply(string num1, string num2) {
        int m = num1.size();
        int n = num2.size();
        vector<int> array(m+n);
        string ans = "";
        for (int i = m-1; i >= 0; i--) {
            for (int j = n-1; j >= 0; j--) {
                int index1 = i + j;
                int index2 = i + j + 1;
                int sum = (num1[i]-'0')*(num2[j]-'0')+array[index2];
                array[index1] += sum / 10;
                array[index2] = sum % 10;
            }
        }

        for (auto v:array) {
            if (ans.size() != 0 || v != 0)
                ans  += to_string(v);
        }

        return ans.size() == 0?"0":ans;
        
    }
};

File: Multiply Strings.py
class Solution(object):
    def multiply(self, num1, num2):
        """
        :type num1: str
        :type num2: str
        :rtype: str
        """
        if num1 == '0' or num2 == '0':
            return '0'
        ans = 0
        for i, n1 in enumerate(num2[::-1]):
            pre = 0
            curr = 0
            for j, n2 in enumerate(num1[::-1]):
                multi = (ord(n1) - ord('0')) * (ord(n2) - ord('0'))
                first, second = multi // 10, multi % 10
                curr += (second + pre) * (10 ** j) 
                pre = first
            curr += pre * (10 ** len(num1))
            ans += curr * (10 ** i)
        return str(ans)
        
